diff --git a/src/base/cmd/cmd.c b/src/base/cmd/cmd.c
index a0042443..fe7286b7 100644
--- a/src/base/cmd/cmd.c
+++ b/src/base/cmd/cmd.c
@@ -1161,7 +1161,7 @@ int CmdCommandScanDir( Abc_Frame_t * pAbc, int argc, char **argv )
     struct _finddata_t c_file;
     char * pDirStr = NULL;
     char*  pDirCur = NULL;
-    long   hFile;
+    ABC_PTRINT_T hFile;
     char   c;
 
     Extra_UtilGetoptReset();
@@ -1354,7 +1354,7 @@ void CnfDupFileUnzip( char * pOldName )
 int CmdCommandRenameFiles( Abc_Frame_t * pAbc, int argc, char **argv )
 {
     struct _finddata_t c_file;
-    long   hFile;
+    ABC_PTRINT_T hFile;
     char pNewName[1000];
     char * pDirStr = NULL;
     char * pDirCur = NULL;
@@ -1515,7 +1515,7 @@ usage:
 int CmdCommandLs( Abc_Frame_t * pAbc, int argc, char **argv )
 {
     struct _finddata_t c_file;
-    long   hFile;
+    ABC_PTRINT_T hFile;
     int    fLong = 0;
     int    fOnlyBLIF = 0;
     char   Buffer[25];
@@ -1618,7 +1618,7 @@ usage:
 int CmdCommandScrGen( Abc_Frame_t * pAbc, int argc, char **argv )
 {
     struct _finddata_t c_file;
-    long   hFile;
+    ABC_PTRINT_T hFile;
     FILE * pFile = NULL;
     char * pFileStr = "test.s";
     char * pDirStr = NULL;
diff --git a/src/base/cmd/cmdLoad.c b/src/base/cmd/cmdLoad.c
index accd9440..3987964c 100644
--- a/src/base/cmd/cmdLoad.c
+++ b/src/base/cmd/cmdLoad.c
@@ -114,7 +114,7 @@ Vec_Ptr_t * CmdCollectFileNames()
 {
     Vec_Ptr_t * vFileNames;
     struct _finddata_t c_file;
-    long   hFile;
+    ABC_PTRINT_T hFile;
     if( (hFile = _findfirst( "*.exe", &c_file )) == -1L )
     {
 //        Abc_Print( 0, "No files with extention \"%s\" in the current directory.\n", "exe" );
