diff --git a/src/lib/bitblast/aig/aig_cnf.cpp b/src/lib/bitblast/aig/aig_cnf.cpp
index a9a00c8e..69a19ad6 100644
--- a/src/lib/bitblast/aig/aig_cnf.cpp
+++ b/src/lib/bitblast/aig/aig_cnf.cpp
@@ -10,6 +10,7 @@
 
 #include "bitblast/aig/aig_cnf.h"
 
+#include <cstdlib>
 #include <functional>
 
 namespace bzla::bb {
diff --git a/src/lib/bitblast/aig/aig_manager.cpp b/src/lib/bitblast/aig/aig_manager.cpp
index 878ef276..7d36b76a 100644
--- a/src/lib/bitblast/aig/aig_manager.cpp
+++ b/src/lib/bitblast/aig/aig_manager.cpp
@@ -10,6 +10,8 @@
 
 #include "bitblast/aig/aig_manager.h"
 
+#include <cstdlib>
+
 namespace bzla::bb {
 
 bool
diff --git a/src/main/options.cpp b/src/main/options.cpp
index 0d1e518a..5f3fb97c 100644
--- a/src/main/options.cpp
+++ b/src/main/options.cpp
@@ -2,6 +2,7 @@
 
 #include <bitwuzla/cpp/bitwuzla.h>
 
+#include <algorithm>
 #include <cassert>
 #include <iomanip>
 #include <iostream>
diff --git a/src/parser/smt2/parser.cpp b/src/parser/smt2/parser.cpp
index 3bd26ffc..164a48be 100644
--- a/src/parser/smt2/parser.cpp
+++ b/src/parser/smt2/parser.cpp
@@ -10,6 +10,7 @@
 
 #include "parser/smt2/parser.h"
 
+#include <algorithm>
 #include <iostream>
 
 namespace bzla {
diff --git a/test/unit/api/test_api.cpp b/test/unit/api/test_api.cpp
index c80bb32d..da7c743d 100644
--- a/test/unit/api/test_api.cpp
+++ b/test/unit/api/test_api.cpp
@@ -11,6 +11,7 @@
 #include <bitwuzla/cpp/bitwuzla.h>
 #include <bitwuzla/cpp/parser.h>
 
+#include <algorithm>
 #include <chrono>
 #include <fstream>
 
