diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index b51e10a1..c1f8932a 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -192,25 +192,29 @@ target_link_libraries(btoruntrace boolector)
 #-----------------------------------------------------------------------------#
 # btorimc
 
-set(btorimc_src_files
-  btoribv.cc
-  btorimc.cc
-)
-add_executable (btorimc ${btorimc_src_files})
-target_link_libraries(btorimc boolector)
+if(IS_WINDOWS_BUILD)
+  message(STATUS "Skipping btorimc for Windows builds")
+else()
+  set(btorimc_src_files
+    btoribv.cc
+    btorimc.cc
+  )
+  add_executable (btorimc ${btorimc_src_files})
+  target_link_libraries(btorimc boolector)
+endif()
 
 #-----------------------------------------------------------------------------#
 # Static binaries
 
 if(NOT BUILD_SHARED_LIBS AND NOT APPLE)
   set(BTOR_STATIC_FLAGS "-static")
-  set(STATIC_TARGETS boolector-bin btormc btoruntrace btorimc)
+  set(STATIC_TARGETS boolector-bin btormc btoruntrace)
 
   if(HAVE_NO_EXPORT_DYNAMIC)
     set(BTOR_STATIC_FLAGS "${BTOR_STATIC_FLAGS} -Wl,--no-export-dynamic")
   endif()
   if(NOT IS_WINDOWS_BUILD)
-    set(STATIC_TARGETS ${STATIC_TARGETS} btormbt)
+    set(STATIC_TARGETS ${STATIC_TARGETS} btormbt btorimc)
   endif()
   set_target_properties(${STATIC_TARGETS}
     PROPERTIES
