diff --git a/configure.sh b/configure.sh
index 21a444082..8b30c8b14 100755
--- a/configure.sh
+++ b/configure.sh
@@ -23,6 +23,7 @@ General options;
   --best                   turn on dependencies known to give best performance
   --gpl                    permit GPL dependencies, if available
   --win64                  cross-compile for Windows 64 bit
+  --win64-native           natively compile for Windows 64 bit
   --ninja                  use Ninja build system
 
 
@@ -146,6 +147,7 @@ ubsan=default
 unit_testing=default
 valgrind=default
 win64=default
+win64_native=default
 
 language_bindings_java=default
 language_bindings_python=default
@@ -238,6 +240,9 @@ do
     --win64) win64=ON;;
     --no-win64) win64=OFF;;
 
+    --win64-native) win64_native=ON;;
+    --no-win64-native) win64_native=OFF;;
+
     --ninja) ninja=ON;;
 
     --glpk) glpk=ON;;
@@ -387,6 +392,9 @@ cmake_opts=""
   && cmake_opts="$cmake_opts -DENABLE_GPL=$gpl"
 [ $win64 != default ] \
   && cmake_opts="$cmake_opts -DCMAKE_TOOLCHAIN_FILE=../cmake/Toolchain-mingw64.cmake"
+# Because 'MSYS Makefiles' has a space in it, we set the variable vs. adding to 'cmake_opts'
+[ $win64_native != default ] \
+  && [ $ninja == default ] && export CMAKE_GENERATOR="MSYS Makefiles"
 [ $ninja != default ] && cmake_opts="$cmake_opts -G Ninja"
 [ $muzzle != default ] \
   && cmake_opts="$cmake_opts -DENABLE_MUZZLE=$muzzle"
diff --git a/src/expr/mkexpr b/src/expr/mkexpr
index c5f12f487..b47c66753 100755
--- a/src/expr/mkexpr
+++ b/src/expr/mkexpr
@@ -14,6 +14,9 @@
 #
 # Output is to standard out.
 #
+# Required to disable this option for bash >=5.2 to avoid automatically
+# replacing & by the substituted text.
+shopt | grep -q '^patsub_replacement\b' &&  shopt -u patsub_replacement
 
 copyright=2010-2014
 
diff --git a/src/expr/mkkind b/src/expr/mkkind
index fbf37eff4..aabc70afe 100755
--- a/src/expr/mkkind
+++ b/src/expr/mkkind
@@ -13,6 +13,9 @@
 #
 # Output is to standard out.
 #
+# Required to disable this option for bash >=5.2 to avoid automatically
+# replacing & by the substituted text.
+shopt | grep -q '^patsub_replacement\b' &&  shopt -u patsub_replacement
 
 copyright=2010-2014
 
diff --git a/src/expr/mkmetakind b/src/expr/mkmetakind
index e2a733ec8..e56d076d7 100755
--- a/src/expr/mkmetakind
+++ b/src/expr/mkmetakind
@@ -16,6 +16,9 @@
 #
 # Output is to standard out.
 #
+# Required to disable this option for bash >=5.2 to avoid automatically
+# replacing & by the substituted text.
+shopt | grep -q '^patsub_replacement\b' &&  shopt -u patsub_replacement
 
 copyright=2010-2014
 
