diff --git a/src/expr/emptyset.h b/src/expr/emptyset.h
index 55e07747e..250185809 100644
--- a/src/expr/emptyset.h
+++ b/src/expr/emptyset.h
@@ -19,6 +19,7 @@

 #pragma once

+#include <cstddef>
 #include <iosfwd>

 namespace CVC4 {
diff --git a/src/expr/expr_iomanip.h b/src/expr/expr_iomanip.h
index f8464392a..985a3ca2a 100644
--- a/src/expr/expr_iomanip.h
+++ b/src/expr/expr_iomanip.h
@@ -19,6 +19,7 @@
 #ifndef CVC4__EXPR__EXPR_IOMANIP_H
 #define CVC4__EXPR__EXPR_IOMANIP_H

+#include <cstddef>
 #include <iosfwd>

 namespace CVC4 {
diff --git a/src/util/regexp.h b/src/util/regexp.h
index 180bb0c32..25c5c9ad2 100644
--- a/src/util/regexp.h
+++ b/src/util/regexp.h
@@ -17,6 +17,7 @@
 #ifndef CVC4__UTIL__REGEXP_H
 #define CVC4__UTIL__REGEXP_H

+#include <cstddef>
 #include <cstdint>
 #include <iosfwd>

